### PR TITLE
NewsML on Plone 5

### PR DESCRIPTION
**Pull request by pbauer**

### CODE REVIEW

**Comment by pbauer** [comment]
test: http://jenkins.plone.org/job/pull-request-5.0/305/

**Comment by jensens** [comment]
one test fails

**Comment by hvelarde** [comment]
@rodfersou FYI

**Comment by rodfersou** [comment]
:+1:

**Comment by jensens** [comment]
rebase needed

**Comment by hvelarde** [comment]
rebase done!

guys I really don't understand how testing works; can someone point me to the Jenkins job? what happened with mr.roboto?

**Comment by pbauer** [comment]
1. you login to http://jenkins.plone.org/
2. go to "Pull Request 5.0" (http://jenkins.plone.org/job/pull-request-5.0/)
3. click "Build with parameters" (http://jenkins.plone.org/job/pull-request-5.0/build?delay=0sec)
4. paste the url of the pull-request
5. click "Build"
6. paste the build-url here.

**Comment by rodfersou** [comment]
@pbauer I don't know if it is just with me, but the "Build with parameters" url don't show here.

but the [url you posted](http://jenkins.plone.org/job/pull-request-5.0/build?delay=0sec) works

**Comment by rodfersou** [comment]
@pbauer sorry, I found now the url (I feel blind now :) )

**Comment by hvelarde** [comment]
so is a manual process now: http://jenkins.plone.org/job/pull-request-5.0/321/

**Comment by rodfersou** [comment]
How can I run tests on a specific group? troup `plone_app_tests` give an exception, but when I run
```
buildout.coredev $ ./bin/test -s Products.CMFPlone --layer Integration
```

It didn't pass this test (I added a breakpoint there)

**Comment by rodfersou** [comment]
The problem happen in this line: https://github.com/plone/Products.CMFPlone/blob/NewsML/Products/CMFPlone/tests/testSyndication.py#L232

**Comment by rodfersou** [comment]
New build http://jenkins.plone.org/job/pull-request-5.0/322/

**Comment by rodfersou** [comment]
I'm not sure if Jenkins worked fine, the [first build had 92 packages](http://jenkins.plone.org/job/pull-request-5.0/321/console) while the [last one had just 3 packages](http://jenkins.plone.org/job/pull-request-5.0/322/console)

**Comment by hvelarde** [comment]
@pbauer @jensens @vangheem green light!

File: Products/CMFPlone/browser/syndication/views.py [hunk 1]
```diff
@@ -50,6 +55,62 @@ def __call__(self):
...
+    def newsml_enabled(self, raise404=False):
+        if not self.newsml_allowed():
+            if raise404:
+                raise NotFound
+            else:
+                return False
+        else:
+            return True
+
+    def __call__(self):
```
Comment (vangheem) [inline Products/CMFPlone/browser/syndication/views.py]: Why is the __call__ method overridden here? I don't think newsml_allowed and newsml_enabled are then necessary.

**Comment by vangheem** [comment]
made some changes, one more test run: http://jenkins.plone.org/job/pull-request-5.0/326/

**Comment by pbauer** [comment]
Aaaaand it red again.

**Comment by rodfersou** [comment]
@vangheem the problem is with the methods you removed `newsml_allowed`, you should put it back or remove the test.

```
Error in test test_newsml_allowed (Products.CMFPlone.tests.testSyndication.TestNewsML)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/home/jenkins/workspace/pull-request-5.0/src/Products.CMFPlone/Products/CMFPlone/tests/testSyndication.py", line 299, in test_newsml_allowed
    self.assertEqual(view.newsml_allowed(), True)
AttributeError: 'SimpleViewClass from /home/jenkins/workspace/pull-' object has no attribute 'newsml_allowed'



Error in test test_newsml_allowed_not_syndicatable (Products.CMFPlone.tests.testSyndication.TestNewsML)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/home/jenkins/workspace/pull-request-5.0/src/Products.CMFPlone/Products/CMFPlone/tests/testSyndication.py", line 294, in test_newsml_allowed_not_syndicatable
    view = self.folder.file.restrictedTraverse('@@newsml.xml')
  File "/home/jenkins/.buildout/eggs/Zope2-2.13.23-py2.7.egg/OFS/Traversable.py", line 317, in restrictedTraverse
    return self.unrestrictedTraverse(path, default, restricted=True)
  File "/home/jenkins/.buildout/eggs/Zope2-2.13.23-py2.7.egg/OFS/Traversable.py", line 285, in unrestrictedTraverse
    raise e
   - __traceback_info__: ([], '@@newsml.xml')
AttributeError: @@newsml.xml



Error in test test_newsml_allowed_site_disabled (Products.CMFPlone.tests.testSyndication.TestNewsML)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/home/jenkins/workspace/pull-request-5.0/src/Products.CMFPlone/Products/CMFPlone/tests/testSyndication.py", line 304, in test_newsml_allowed_site_disabled
    self.assertEqual(view.newsml_allowed(), False)
AttributeError: 'SimpleViewClass from /home/jenkins/workspace/pull-' object has no attribute 'newsml_allowed'



Error in test test_newsml_enabled (Products.CMFPlone.tests.testSyndication.TestNewsML)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/home/jenkins/workspace/pull-request-5.0/src/Products.CMFPlone/Products/CMFPlone/tests/testSyndication.py", line 311, in test_newsml_enabled
    self.assertEqual(view.newsml_enabled(), True)
AttributeError: 'SimpleViewClass from /home/jenkins/workspace/pull-' object has no attribute 'newsml_enabled'



Error in test test_newsml_enabled_raises_404 (Products.CMFPlone.tests.testSyndication.TestNewsML)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/home/jenkins/workspace/pull-request-5.0/src/Products.CMFPlone/Products/CMFPlone/tests/testSyndication.py", line 329, in test_newsml_enabled_raises_404
    self.assertRaises(NotFound, view.newsml_enabled, True)
AttributeError: 'SimpleViewClass from /home/jenkins/workspace/pull-' object has no attribute 'newsml_enabled'



Error in test test_newsml_enabled_site_disabled (Products.CMFPlone.tests.testSyndication.TestNewsML)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/home/jenkins/workspace/pull-request-5.0/src/Products.CMFPlone/Products/CMFPlone/tests/testSyndication.py", line 322, in test_newsml_enabled_site_disabled
    self.assertEqual(view.newsml_enabled(), False)
AttributeError: 'SimpleViewClass from /home/jenkins/workspace/pull-' object has no attribute 'newsml_enabled'



Error in test test_not_newsml_enabled (Products.CMFPlone.tests.testSyndication.TestNewsML)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/home/jenkins/workspace/pull-request-5.0/src/Products.CMFPlone/Products/CMFPlone/tests/testSyndication.py", line 316, in test_not_newsml_enabled
    self.assertEqual(view.newsml_enabled(), False)
AttributeError: 'SimpleViewClass from /home/jenkins/workspace/pull-' object has no attribute 'newsml_enabled'
```

**Comment by rodfersou** [comment]
Or rewrite the tests without these methods

**Comment by vangheem** [comment]
running tests again: http://jenkins.plone.org/job/pull-request-5.0/328/

Those methods were essentially copies of base syndication view logic. There are already tests for the base view.

**Comment by rodfersou** [comment]
@pbauer @jensens @vangheem green light again!

**Comment by hvelarde** [comment]
I can't believe it!

**Comment by pbauer** [comment]
:tada: I can't believe that you were sitting on this since march.

**Comment by hvelarde** [comment]
march 2014

**Comment by pbauer** [comment]
brrr

**Pull request merged**